config/config.py

```diff
@@ -1,71 +1,58 @@
 import os
-from typing import Union
-from utils.ordered_yaml import OrderedYaml
 
-ordered_yaml = OrderedYaml()
+from utils.ordered_yaml import OrderedYaml
 
 
 class Config:
-    SLACK = 'slack'
-    TOKEN = 'token'
-    NOTIFICATION_CHANNEL_NAME = 'channel_name'
-    NOTIFICATION_WEBHOOK = 'notification_webhook'
-    WORKFLOWS = 'workflows'
-    CONFIG_FILE_NAME = 'config.yml'
+    _SLACK = 'slack'
+    _AWS = 'aws'
+    _CONFIG_FILE_NAME = 'config.yml'
 
-    def __init__(self, config_dir: str, profiles_dir: str, profile_target: str = None) -> None:
+    def __init__(self, config_dir: str, profiles_dir: str, profile_target: str = None, slack_webhook: str = None,
+                 slack_token: str = None, slack_channel_name: str = None, aws_profile_name: str = None,
+                 aws_access_key_id: str = None, aws_secret_access_key: str = None, s3_bucket_name: str = None):
         self.config_dir = config_dir
         self.profiles_dir = profiles_dir
         self.profile_target = profile_target
-        self.config_dict = self._load_configuration()
+        config = self._load_configuration()
+
+        self.target_dir = config.get('target-path') or os.getcwd()
+
+        self.slack_webhook = slack_webhook or config.get(self._SLACK, {}).get('notification_webhook')
+        self.slack_token = slack_token or config.get(self._SLACK, {}).get('token')
+        self.slack_channel_name = slack_channel_name or config.get(self._SLACK, {}).get('channel_name')
+        self.is_slack_workflow = config.get(self._SLACK, {}).get('workflows', False)
+
+        self.aws_profile_name = aws_profile_name or config.get(self._AWS, {}).get('profile_name')
+        self.aws_access_key_id = aws_access_key_id
```

Comment:
These files are not retrieved from the configuration.
They could either be supplied via the CLI arguments, or read from `~/.aws/credentials`.